eclair-core/src/main/scala/fr/acinq/eclair/Eclair.scala

```diff
@@ -424,6 +422,14 @@ class EclairImpl(appKit: Kit) extends Eclair with Logging {
     Future.foldLeft(commands)(Map.empty[ApiTypes.ChannelIdentifier, Either[Throwable, T]])(_ + _)
   }
 
+  /** Send a request to multiple channels using node ids */
+  private def sendToNodes[T: ClassTag](nodeids: List[PublicKey], request: Any)(implicit timeout: Timeout): Future[Map[ApiTypes.ChannelIdentifier, Either[Throwable, T]]] = {
+    for {
+      channelIds <- (appKit.register ? Symbol("channelsTo")).mapTo[Map[ByteVector32, PublicKey]].map(_.filter(kv => nodeids.contains(kv._2)).keys)
```

Comment:
Note that this has a side-effect compared to calling `Router.GetLocalChannels`: you won't send to channels that are offline.

IIUC that's done on purpose here, because for the case of updating fees, we've stored it in the DB and the channel will update automatically on restore.

But I think it should be mentioned in a comment, otherwise we'll get bitten by this in the future when we use that function hoping that it sends to channels regardless of their state.

Reply:
It's the opposite actually: offline channels will be returned by the `register`, but not by the `router`.

> IIUC that's done on purpose here, because for the case of updating fees, we've stored it in the DB and the channel will update automatically on restore.

It would indeed have been on _restore_, not on _reconnection_, which would have been really annoying. But that's not what happens, because we actually update channels the same way, whether they are `NORMAL`, `OFFLINE` or `SYNCING`. Where it differs is whether or not a `LocalChannelUpdate` will be sent, but that's not our concern anymore with this PR.

> But I think it should be mentioned in a comment, otherwise we'll get bitten by this in the future when we use that function hoping that it sends to channels regardless of their state.

Sending to all channels that correspond to whatever identifier(s) is/are provided is indeed the default expectation, which is why there is no specific comment. I think that would be confusing?

Reply:
> Sending to all channels that correspond to whatever identifier(s) is/are provided is indeed the default expectation, which is why there is no specific comment. I think that would be confusing?

You're right, if it does send to all channels then no need for a comment, I incorrectly thought offline channels were ignored, but looking at the `ChannelRestored` event we're all good.